lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -350,63 +328,90 @@ loadOpsToDistanceAndUse(scf::ForOp forOp) {
   return loadOpToDistAndUse;
 }
 
+static bool loadIsMMAv3(tt::LoadOp loadOp) {
+  if (!loadOp->hasOneUse())
+    return false;
+  auto alloc = dyn_cast<ttg::LocalAllocOp>(*loadOp->getUsers().begin());
+  if (!alloc)
+    return false;
+  auto sharedEnc =
+      alloc.getType().getEncoding().cast<ttg::SharedEncodingAttr>();
+  if (!sharedEnc.getHasLeadingOffset())
+    return false;
+
+  // MMA V3 case.
+  auto newOrder = sharedEnc.getOrder();
+  auto ty = loadOp.getType().cast<RankedTensorType>();
+  auto oldOrder = ttg::getOrder(ty.getEncoding());
+
+  // The operand of MMAv3 is in SharedEncoding and its order should not
+  // be changed after FuseTranspositions Pass. So we only pipeline the
+  // load if the order of the loaded BlockedEncoding is the same as the
+  // order of the SharedEncoding it is converted to.
+  return oldOrder == newOrder;
+}
+
 /// Collect loads to pipeline. Returns true if loads are found to pipeline.
 static bool
-collectOpsToPipeline(scf::ForOp forOp,
-                     llvm::MapVector<Operation *, PipelinedOpInfo> &opInfo,
-                     int numStages, bool &hasMMAV3) {
+collectLoadsToPipeline(scf::ForOp forOp,
```

Comment:
NIT: This function still does collect users of the loads, like dot, right?

Reply:
Ah, you're right.  Reverted the name and fixed the comment.  Thanks.